exploration/number-selection.md

```diff
@@ -92,44 +95,274 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
+>                                          // such locales typically require other keywords
+>                                          // such as two, few, many, and so forth
+> *   {{You have {$var} chances remaining}}
 > ```
 
-Additional options such as `minimumFractionDigits` and others already supported by `:number`
-should also be supported.
+### Selection
+
+When implementing [`MatchSelectorKeys`](spec/formatting.md#resolve-preferences), 
+numeric selectors perform as described below.
+
+- Let `return_value` be a new empty list of strings.
+- Let `operand` be the resolved value of the _operand_.
+  If the `operand` is not a number type, emit a _Selection Error_
+  and return `return_value`.
+  > This might result in the fallback message (logo) being rendered.
+- Let `keys` be a list of strings containing keys to match.
+  (Hint: this list is an argument to `MatchSelectorKeys`)
+- For each string `key` in `keys`:
+   - If the value of `key` does not match the production `number-literal`
+     or is not one of the plural/ordinal keywords, `key` is invalid;
+     emit a _Selection Error_.
+     `key` is not added to the `return_value`.
+   - If the value of `key` matches the production `number-literal`:
+     - If the parsed value of `key` is an [exact match](#determining-exact-literal-match)
+       of the value of the `operand`, then `key` matches the selector.
+       Add `key` to the front of the `return_value` list.
+   - Else, if the value of `key` is a keyword:
+      - Let `keyword` be a string which is the result of [rule selection](#rule-selection).
+      - If `keyword` equals `key`, then `key` matches the selector.
+        Append `key` to the end of the `return_value` list.
+   - Else, `key` is not a match.
+     Do not add `key` to `return_value`
+- Return `return_value`
+
+### Plural/Ordinal Keywords
+The _plural/ordinal keywords_ are: `zero`, `one`, `two`, `few`, `many`, and
+`other`.
+The value `other` is both a valid output of [rule selection](#rule-selection)
+and serves a default value.
+
+### Rule Selection
+
+If the option `select` is set to `exact`, rule-based selection is not used.
+Return the empty string.
+
+> [!NOTE]
+> Since keys cannot be the empty string in a numeric selector, returning the
+> empty string disables keyword selection
+
+If the option `select` is set to `plural`, selection should be based on CLDR plural rule data
+of type `cardinal`. See [charts](https://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html)
+for examples.
+
+If the option `select` is set to `ordinal`, selection should be based on CLDR plural rule data
+of type `ordinal`. See [charts](https://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html)
+for examples.
+
+Apply the rules defined by CLDR to the resolved value of the operand and return the resulting keyword.
+If no rules match, return `other`.
```

Comment:
```suggestion
Apply the rules defined by CLDR to the resolved value of the operand and the function options,
and return the resulting keyword.
If no rules are available for the requested locale or locales, return `other`.
```

Reply:
This would be incorrect. If the specific locale requested has no rules, the rules used will fall back up the locale hierarchy, ultimately to the root locale. There is always a root locale set of resources.

My text "If no rules match..." covers any errors in CLDR data.

Reply:
Ah. I had presumed we could assume the CLDR data to be infallible, but that is indeed a risky supposition.

Reply:
Well, nothing is infallible.

A more accurate picture is that CLDR does not have data for all 7,000ish languages, so some languages will fall back to root.
And of course, we add locales over time, so a language that falls back to root now, may change in the future; and every now and then we find that we need to tweak plural rules and/or number formats in existing locale data because we didn't fully capture usage (or even, sometimes, that usage changes over time).

Reply:
As noted in a previous comment, many locales produce `other` and the `other` resource in `root` is not very helpful.

I also included the last statement to guard against implementations returning nothing.

The other fix seems valid.

```suggestion
Apply the rules defined by CLDR to the resolved value of the operand and the function options,
and return the resulting keyword.
If no rules match, return `other`.
```